search/search_param_types.go

```diff
@@ -464,6 +507,25 @@ func (s SearchParamInfo) clone() SearchParamInfo {
 	return newInfo
 }
 
+func createReverseChainedQueryInfo(resource, modifier, value string) SearchParamInfo {
+	// First split the modifier (e.g. "Observation:subject:code")
+	// to get the resources being referenced and a SearchParam to search.
+	// the parts are [fromResource, refField, searchField]
+	parts := strings.SplitN(modifier, ":", 3)
+	if len(parts) != 3 {
+		panic(createInternalServerError("MSG_PARAM_INVALID", fmt.Sprintf("Parameter \"%s\" content is invalid", "_has")))
+	}
+	refInfo, ok := SearchParameterDictionary[parts[0]][parts[1]]
+	if !ok {
+		panic(createInvalidSearchError("SEARCH_NONE", fmt.Sprintf("Error: no processable search found for %s search parameters \"%s\"", resource, "_has")))
+	}
+	revChainInfo := refInfo.clone()
+	revChainInfo.Resource = resource
+	revChainInfo.Name = "_has"
+	revChainInfo.Targets = []string{resource} // We already know the target
```

Comment:
The fact that both the `revChainInfo.Resource` and the `revChainInfo.Targets` are set to the `resource` really confused me, but I think my brain is tired.  Since the `_has` modifier makes everything backwards, it's probably right.  But man... you have to admit it's kind of confusing.

I was also confused that when you're done here, `revChainInfo` has no reference at all to the referenced resource type (e.g., "Observation"), the reference field (e.g., "subject"), or the referenced query param (e.g., "code").  It feels like it's getting tossed.  You might want to put in a comment indicating that the `modifier` will be added to the `revChainInfo` in the `Params()` function.  Or you might not.  I don't know.